src/Models/ArduinoDeviceBase.ts

```diff
@@ -121,8 +122,54 @@ export abstract class ArduinoDeviceBase implements Device {
   abstract async configDeviceSettings(): Promise<boolean>;
 
   abstract async load(): Promise<boolean>;
+
+
   abstract async create(): Promise<boolean>;
 
+  async createCore(board: Board|undefined, templateFiles: TemplateFileInfo[]):
+      Promise<boolean> {
+    const deviceFolderPath = this.deviceFolder;
+
+    const scaffoldType = ScaffoldType.Local;
+    if (!await FileUtility.directoryExists(scaffoldType, deviceFolderPath)) {
+      throw new Error('Unable to find the device folder inside the project.');
+    }
+    if (!board) {
+      throw new Error('Unable to find the board in the config file.');
+    }
+
+    const plat = await IoTWorkbenchSettings.getPlatform();
+
+    await IoTWorkbenchProjectBase.generateIotWorkbenchProjectFile(
+        scaffoldType, this.deviceFolder);
+
+    for (const fileInfo of templateFiles) {
+      if ((fileInfo.fileName.endsWith('macos.json') ||
+           fileInfo.fileName.endsWith('win32.json'))) {
+        if (fileInfo.fileName.endsWith('macos.json') && (plat === 'darwin')) {
+          await this.generateCppPropertiesFile(scaffoldType, board, fileInfo);
+        } else if (
+            fileInfo.fileName.endsWith('win32.json') && (plat === 'win32')) {
+          await this.generateCppPropertiesFile(scaffoldType, board, fileInfo);

```

Comment:
what's the different between drawin and win32 here? both invoke 'await this.generateCppPropertiesFile(scaffoldType, board, fileInfo);'

Reply:
Fixed.